### PR TITLE
Fix: Project did not migrate to `pyproject.toml`

### PR DESCRIPTION
**Pull request by hoh**
The project still used `setup.py` and `setup.cfg`.

This branch:
1. Migrates to pyproject.toml and `hatch`.
2. Updates workflows accordingly.
3. Changes README from reStructuredText to the easier Markdown.
4. Updates dependencies.

This also fixes multiple issues and bugs in the code. Do not squash.

Reviewers may want to look at commits individually.

### CODE REVIEW

**Comment by hoh** [comment]
Requires a new release of aleph-sdk-python

File: README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+> brew tap cuber/homebrew-libsecp256k1
+> brew install libsecp256k1
+
+## Installation
+
+Using pip and [PyPI](https://pypi.org/project/aleph-client/):
+
+> pip install aleph-client
+
+## Installation for development
```
Comment (Psycojoker) [inline README.md:31]: Shouldn't we also add instructions regarding `hatch` here?

I see some of them for the python SDK https://github.com/aleph-im/aleph-sdk-python?tab=readme-ov-file#installation-for-development

Comment (hoh) [reply README.md:47]: Good point. Added via commit https://github.com/aleph-im/aleph-client/pull/213/commits/e97df3b8e0401f0f16dc6daa1ef10787d156c696

**Comment by Psycojoker** [comment]
We are waiting for a new version of the SDK before merging this one cc @hoh

File: src/aleph_client/commands/program.py [hunk 2]
```diff
@@ -200,11 +200,11 @@ async def update(
     async with AuthenticatedAlephHttpClient(
         account=account, api_server=sdk_settings.API_HOST
     ) as client:
-        program_message: ProgramMessage = client.get_message(
+        program_message: ProgramMessage = await client.get_message(
             item_hash=item_hash, message_type=ProgramMessage
         )
         code_ref = program_message.content.code.ref
-        code_message: StoreMessage = client.get_message(
+        code_message: StoreMessage = await client.get_message(
```
[the remark this answers is not in the document]
Comment (hoh) [reply src/aleph_client/commands/program.py]: Many tests were missing.

**Pull request merged**